src/Symfony/Bridge/Doctrine/composer.json

```diff
@@ -22,6 +22,7 @@
         "symfony/polyfill-ctype": "~1.8",
         "symfony/polyfill-mbstring": "~1.0",
         "symfony/polyfill-php80": "^1.15",
+        "symfony/uid": "^5.1",
```

Comment:
this should be added to `require-dev`
the constructors of the above classes could then do a runtime check + throw if the component is missing (see similar LogicException in the codebase when a dep is missing)

Reply:
The Base Type class has a private construct https://github.com/doctrine/dbal/blob/e7c4149a13f1267380a8061452816a1a33909dc1/lib/Doctrine/DBAL/Types/Type.php#L134, therefor I can not do the runtime check you are suggesting.

I'm going to implement IdGenerators tomorrow.

Reply:
well, if we go for my proposal of a separate package, the issue of the optional dependency is solved, btw.